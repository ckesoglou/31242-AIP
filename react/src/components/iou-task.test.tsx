import React from "react";
import { shallow, mount } from "enzyme";
import IouTask from "./iou-task";

// simple, static component
describe("<IouTask />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<IouTask details="Finish my homework for me" />);

    expect(wrapper).toMatchSnapshot();
  });

  it("should shorten task detail when it is too long", () => {
    const wrapper = mount(
      <IouTask details="Clean the fridge asdf sadf asdf asdf sdaf dsaf sadf sadf SOmething extra that shouldn't be seen" />
    );

    let value = wrapper.find("#taskDetailShort").text();
    expect(value).toContain("...");
  });
});
