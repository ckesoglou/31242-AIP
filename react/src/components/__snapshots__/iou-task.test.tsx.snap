// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IouTask /> should render correctly 1`] = `
<div
  className=""
  id="task"
>
  <WithStyles(ForwardRef(Typography))>
    Finish my homework for me
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Popover))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    id="popUpMargin"
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      id="taskDetailPopUp"
    >
      Task Description
      <WithStyles(ForwardRef(Divider))
        id="taskPopUpDivider"
      />
      Finish my homework for me
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Popover))>
</div>
`;
